src/components/ui/InputWithIcon/InputWithIcon.tsx

```diff
@@ -0,0 +1,38 @@
+import { forwardRef } from "react";
+
+import { cn } from "@/utils/helper";
+import { IconPosition } from "@/models/enums/ButtonVariant";
+import { IInputWithIconProps } from "@/components/ui/InputWithIcon/InputWithIcon.d";
+
+const InputWithIcon = forwardRef<HTMLInputElement, IInputWithIconProps>(
+  (props, ref) => {
+    const { icon, iconPosition, className, ...rest } = props;
+
+    return (
+      <span className="relative">
+        {!!icon && iconPosition === IconPosition.LEFT && (
+          <span className="absolute top-1/2 transform -translate-y-1/2 translate-x-4">
+            {icon}
+          </span>
+        )}
+        <input
```

Comment:
What kind of `input` it is? As per AODA, guideline, every `input` should have `type` associated with it.